packages/composer/lib/proxy.js

```diff
@@ -7,31 +7,21 @@ const fp = require('fastify-plugin')
 const kITC = Symbol.for('plt.runtime.itc')
 
 async function resolveServiceProxyParameters (service) {
-  let {
-    origin,
-    proxy: { prefix }
-  } = service
+  // Get meta information from the service, if any, to eventually hook up to a TCP port
+  const meta = (await globalThis[kITC]?.send('getServiceMeta', service.id))?.composer ?? {}
+
+  const origin = meta.tcp ? meta.url : service.origin
 
+  let prefix = service.proxy?.prefix ?? meta.prefix ?? service.id
   if (prefix.endsWith('/')) {
     prefix = prefix.slice(0, -1)
   }
 
   let rewritePrefix = ''
   let internalRewriteLocationHeader = true
 
-  // Get meta information from the service, if any, to eventually hook up to a TCP port
-  const meta = (await globalThis[kITC]?.send('getServiceMeta', service.id))?.composer ?? {}
-
-  if (meta.tcp) {
-    origin = meta.url
-  }
-
-  if (meta.prefix) {
-    prefix = meta.prefix
-  }
-
   if (meta.wantsAbsoluteUrls) {
-    rewritePrefix = prefix
+    rewritePrefix = meta.prefix ?? service.id
```

Comment:
I think this is wrong. If the application wants no absoluteUrl we strip the prefix completely.

Reply:
I don't think so. Example: frontend app has a basePath equal to `/app1/frontend`. It's exposed through a composer via `/frontend`. To make it work:
`prefix` should be equal to `/frontend`
`rewritePrefix` should be equal to `/app1/frontend`

If we assign prefix to the rewritePrefix, the `/app` prefix part will be trimmed and it will not work.

Reply:
Btw I'm not sure why you are asking about the case when `wantsAbsoluteUrls` is false. I didn't change the behavior for this case. If `wantsAbsoluteUrls` is false the rewritePrefix still equals an empty string.

Reply:
I should not review code at 21:55.
It looks fine :)